pandas/core/internals/blocks.py

```diff
@@ -2557,6 +2557,11 @@ def new_block(
     # - check_ndim/ensure_block_shape already checked
     # - maybe_coerce_values already called/unnecessary
     klass = get_block_type(values.dtype)
+
+    # Old pandas (<=1.3.0) will call this function with placements
+    # that aren't necessarily BlockPlacements when unpickling
+    if not isinstance(placement, BlockPlacement):
```

Comment:
also maybe deprecate reading of such-old pickles?

Reply:
Good idea on deprecating reading old pickles, I'll do that as a follow-up (1.3.5 is 2 years old at this point I think).

How would I change ``__setstate__``? I'm not familiar at all with pickle stuff.

I thought ``__reduce__`` calls ``new_block`` directly in 1.3.5.

https://github.com/pandas-dev/pandas/blob/66e3805b8cabe977f40c05259cc3fcf7ead5687d/pandas/_libs/internals.pyx#L485-L493

Reply:
yah, `__reduce__` sounds like the right place for this

Reply:
Hm, the old ``__reduce__`` (from 1.3.5) is getting called though, which is leading to the problem, since it directly calls ``new_block``.

The newer reduce goes through ``_unpickle_block``, which does the conversion to ``BlockPlacement``.

Are we OK just taking the perf hit for now?

Reply:
Might need patching in compat.pickle_compat then.  That is always a PITA.

i think id be against taking the perf hit for something so old since it would be hit a zillion times.